src/Symfony/Component/ErrorHandler/ErrorRenderer/HtmlErrorRenderer.php

```diff
@@ -66,9 +66,12 @@ public function __construct($debug = false, string $charset = null, $fileLinkFor
      */
     public function render(\Throwable $exception): FlattenException
     {
-        $exception = FlattenException::createFromThrowable($exception, null, [
-            'Content-Type' => 'text/html; charset='.$this->charset,
-        ]);
+        $headers = ['Content-Type' => 'text/html; charset='.$this->charset];
+        if (\is_bool($this->debug) ? $this->debug : ($this->debug)($exception)) {
+            $headers['X-Debug-Exception'] = $exception->getMessage();
```

Comment:
Another option would be to add X-Debug-Exception header only for 5xx exceptions (instead of any kind of exception).